server/consumer.go

```diff
@@ -1036,7 +1036,10 @@ func (o *consumer) processAck(_ *subscription, c *client, subject, reply string,
 		o.ackMsg(sseq, dseq, dc)
 	case bytes.HasPrefix(msg, AckNext):
 		o.ackMsg(sseq, dseq, dc)
+		phdr := c.pa.hdr
```

Comment:
Let's add a comment here so we understand why we are doing this:
```
// processNextMsgReq can be invoked from an internal subscription or from here.
// Therefore, it has to call msgParts(), so we can't simply pass msg[len(AckNext):]
// with current c.pa.hdr because it would cause a panic. We will save the current
// c.pa.hdr value and disable headers before calling processNextMsgReq and then
// restore so that we don't mess with the calling stack in case it is used somewhere else.
c.pa.hdr = -1
...
```

Reply:
Thanks, added the comment.